core/src/com/unciv/logic/city/IConstruction.kt

```diff
@@ -129,7 +129,7 @@ class RejectionReasons: HashSet<RejectionReason>() {
 
 // TODO: Put a wrapper class around this containing the errorMessage, so that we don't
 // change the value of a enum constant sometimes.
-enum class RejectionReason(val shouldShow: Boolean, var errorMessage: String) {
+enum class RejectionReason(var shouldShow: Boolean, var errorMessage: String) {
```

Comment:
I would like to note that changing the `shouldShow` variable for one instance of the enum, changes it for all instances iirc. I.e., every unit with the `CannotBuildUnits` unique will be shown, with the same error message as the one that is blocked by having to little happiness. This is probably not expected behaviour.

Reply:
Oh
That makes sense in retrospect but is not what I want

Reply:
I think I warned long ago that overwriting vars on enums would come back to haunt us - it's not 'all instances' but 'a singleton per enum entry'.